### PR TITLE
MM-20452 Migrate installed OAuth apps component to TypeScript

### PR DESCRIPTION
**Pull request by lmammino**
**Warning: Noob ahead**

> 👋      Hello, this is my first PR to any of the Mattermost projects, so please be lenient with your review 😛

#### Summary

This PR addresses #16122 and migrates the OAuth apps component to TypeScript

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/16122

JIRA ticket: https://github.com/mattermost/mattermost-server/issues/16122#:~:text=https%3A//mattermost.atlassian.net/browse/MM-20452

#### Notes

- Some tests seem to be failing on my machine, but I have the same failures also on current `master`.

### CODE REVIEW

File: components/integrations/installed_oauth_apps/installed_oauth_apps.jsx
Comment (hmhealey) [inline components/integrations/installed_oauth_apps/installed_oauth_apps.jsx:10]: This can be reverted. Webpack knows to look for the index.ts when you import a folder.

**Review by hmhealey — CHANGES_REQUESTED** [review]

File: components/integrations/installed_oauth_app/installed_oauth_app.tsx
Comment (lmammino) [reply components/integrations/installed_oauth_app/installed_oauth_app.tsx:18]: Awesome, thanks! I'll update this one shortly!

File: components/integrations/installed_oauth_app/installed_oauth_app.tsx
Comment (lmammino) [reply components/integrations/installed_oauth_app/installed_oauth_app.tsx:45]: Thank you! As above, I'll update this one

File: components/integrations/installed_oauth_apps/installed_oauth_apps.jsx
Comment (lmammino) [reply components/integrations/installed_oauth_apps/installed_oauth_apps.jsx:10]: Oh yes, great catch :) thanks. I'll fix shortly

File: components/integrations/installed_oauth_app/installed_oauth_app.tsx
Comment (lmammino) [reply components/integrations/installed_oauth_app/installed_oauth_app.tsx]: This one turned out to be interesting. In the component we only use `team.name` and therefore all the tests are modelled not to specify all the other mandatory fields in a `Team` instance. I ended up using a `Partial<Team>` for the type hint here, but, to be honest, I am not super happy with it. What do you think? Should we construct a more complete `Team` instance in the tests?

File: components/integrations/installed_oauth_apps/installed_oauth_apps.jsx
Comment (lmammino) [inline components/integrations/installed_oauth_apps/installed_oauth_apps.jsx:10]: Note here: if I don't put the trailing `/`, `eslint` complains about the following:

> Casing of ../installed_oauth_app does not match the underlying filesystem.

File: components/integrations/installed_oauth_app/installed_oauth_app.tsx
Comment (mickmister) [reply components/integrations/installed_oauth_app/installed_oauth_app.tsx]: > Should we construct a more complete `Team` instance in the tests?

I would think that is what is done in other tests. They may be importing it from static data, or possibly using `as Team` in the tests, like `const team = {name: 'The Team'} as Team;`

File: components/integrations/installed_oauth_apps/installed_oauth_apps.jsx
Comment (mickmister) [reply components/integrations/installed_oauth_apps/installed_oauth_apps.jsx:10]: I'm wondering if circle ci will also complain about this if you make a commit without the slash. If circle ci says it's fine, there may be an issue with your eslint setup. If circle ci complains about it, there's another problem to solve here.

**Review by mickmister — APPROVED** [review]
Changes LGTM :+1: Thanks @lmammino

File: components/integrations/installed_oauth_app/installed_oauth_app.tsx
Comment (lmammino) [reply components/integrations/installed_oauth_app/installed_oauth_app.tsx]: Thanks for proving good options @mickmister. I'll check the other tests and fallback to `as Team` if I can't find anything useful there.

File: components/integrations/installed_oauth_apps/installed_oauth_apps.jsx
Comment (lmammino) [reply components/integrations/installed_oauth_apps/installed_oauth_apps.jsx:10]: Don't you have the same issue in your ESLint setup? I am quite sure that my setup will only use what's in the project eslint config...

With that being said, I can easily try and see what happens in CircleCI...

File: components/integrations/installed_oauth_app/installed_oauth_app.tsx
Comment (hmhealey) [reply components/integrations/installed_oauth_app/installed_oauth_app.tsx]: Yeah, I think we mostly use `as Team`. We don't have any good tooling to generate teams, so just providing the fields that actually have an impact on the tests should work well enough.

File: components/integrations/installed_oauth_apps/installed_oauth_apps.jsx
Comment (hmhealey) [reply components/integrations/installed_oauth_apps/installed_oauth_apps.jsx]: CI is passing, so that's strange if you're still getting an issue locally. I know the configuration of imports with ESLint vs other things is a bit painful, but usually relative paths are the thing that actually works well.

Your PR looks otherwise good to merge. I'd be totally fine taking it if it passes CI unless you still wanted to look at it. Otherwise, we can just hope it's an intermittent ESLint problem

Comment (lmammino) [reply components/integrations/installed_oauth_apps/installed_oauth_apps.jsx]: If all look good now and there are no other standing comments, I'd say "let's merge" :)

**Review by hmhealey — APPROVED** [review]

**Comment by lmammino** [comment]
Is there anything i can do to help this one getting merged?

**Review by jgilliam17 — APPROVED** [review]
Thank you @lmammino for your contribution and apologize for the delay in getting to this PR.
I've had some trouble setting up OAuth on the test server, this is good to merge - QA will test after merge.

**Comment by mickmister** [comment]
@lmammino Heads up that there are some changes that need to be made to satisfy the linter before this can be merged:

```
/root/mattermost/mattermost-webapp/components/integrations/installed_oauth_app/installed_oauth_app.tsx
  33:25  error  Expected a semicolon  @typescript-eslint/member-delimiter-style
  38:24  error  Expected a semicolon  @typescript-eslint/member-delimiter-style
  40:25  error  Expected a semicolon  @typescript-eslint/member-delimiter-style
  45:86  error  Expected a semicolon  @typescript-eslint/member-delimiter-style
  50:44  error  Expected a semicolon  @typescript-eslint/member-delimiter-style
  55:28  error  Expected a semicolon  @typescript-eslint/member-delimiter-style
  59:26  error  Expected a semicolon  @typescript-eslint/member-delimiter-style
  60:26  error  Expected a semicolon  @typescript-eslint/member-delimiter-style
```

**Comment by lmammino** [comment]
Hey @mickmister, just updated the PR. Hopefully this will go green 🤞

**Comment by lmammino** [comment]
🟢  now! 

🥳

**Comment by hmhealey** [comment]
Thanks again for the PR!

**Pull request merged**
